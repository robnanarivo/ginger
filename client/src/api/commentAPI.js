import axios from 'axios';
import domain from '../constants/profile';

const createComment = async (comment) => {
  const token = sessionStorage.getItem('userToken');
  try {
    const config = {
      authorization: `Bearer ${token}`,
    };
    const res = await axios.post(`${domain}/api/comments/`, comment, { headers: config });
    return res.data;
  } catch (err) {
    return err.response;
  }
};

const deleteComment = async (commentId) => {
  const token = sessionStorage.getItem('userToken');
  try {
    const config = {
      authorization: `Bearer ${token}`,
    };
    const res = await axios.delete(`${domain}/api/comments/${commentId}`, { headers: config });
    return res.data;
  } catch (err) {
    return err.response;
  }
};

const updateComment = async (commentId, comment) => {
  const token = sessionStorage.getItem('userToken');
  try {
    const config = {
      authorization: `Bearer ${token}`,
    };
    const res = await axios.put(`${domain}/api/comments/${commentId}`, comment, { headers: config });
    return res.data;
  } catch (err) {
    if (err.response) {
      return err.response.data;
    }
    return { error: 'Cannot connect to server' };
  }
};

const getPost = async (postId) => {
  try {
    const res = await axios.get(`${domain}/api/posts/${postId}`);
    const { data } = res;
    const { content } = data;
    if (content.audio) {
      Object.assign(content, { audio: `${domain}/api/multimedia/${content.audio}` });
    }
    if (content.video) {
      Object.assign(content, { video: `${domain}/api/multimedia/${content.video}` });
    }
    if (content.image) {
      Object.assign(content, { image: `${domain}/api/multimedia/${content.image}` });
    }
    return data;
  } catch (err) {
    return err.response;
  }
};

export {
  createComment,
  deleteComment,
  updateComment,
  getPost,
};
