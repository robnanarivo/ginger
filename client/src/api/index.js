import {
  getListOfGroups,
  getGroupData,
  updateGroup,
  removeAdmin,
  getGroupAdmins,
  createGroup,
} from './fetcher';
import login from './login';
import register from './register';
import { getNotifications, updateNotification } from './notificationAPI';
import getUsers from './user';
import { updateProfile, getProfile } from './profileAPI';
import {
  createComment, deleteComment, updateComment, getPost,
} from './commentAPI';

export {
  getGroupData,
  getListOfGroups,
  updateGroup,
  getGroupAdmins,
  removeAdmin,
  login,
  register,
  createGroup,
  getNotifications,
  updateNotification,
  getUsers,
  updateProfile,
  getProfile,
  createComment,
  deleteComment,
  updateComment,
  getPost,
};
