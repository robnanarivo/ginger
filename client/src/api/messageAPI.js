import axios from 'axios';
import domain from '../constants/profile';

const sendMessage = async (message) => {
  try {
    const res = await axios.post(`${domain}/api/messages`, message);
    return res.data;
  } catch (err) {
    return err.response;
  }
};

const getMessageHistory = async (chatter1, chatter2, offset) => {
  const params = {
    chatter1,
    chatter2,
    offset,
    limit: 10,
  };
  try {
    const res = await axios.get(`${domain}/api/messages`, { params });
    return res.data;
  } catch (err) {
    return err.response;
  }
};

const getNewMessage = async (chatter1, chatter2) => {
  const params = {
    chatter1,
    chatter2,
    offset: 0,
    limit: 1,
  };
  try {
    const res = await axios.get(`${domain}/api/messages`, { params });
    return res.data;
  } catch (err) {
    return err.response;
  }
};

export {
  sendMessage,
  getMessageHistory,
  getNewMessage,
};
