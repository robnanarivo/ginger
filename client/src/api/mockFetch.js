export const groupData = {
  topics: ['Sports', 'Universe', 'Science', 'Astronomy'],
  groupInfo: { createOn: 'Feb 20 2008', member: '2.8k' },
  groupId: '11',
  admins: [],
  users: [],
  posts: [
    {
      postId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
      groupId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
      userId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
      content: {
        title: 'Stars are so beautiful',
        text: 'star!',
        image: '/starPic.jpeg',
        audio: '',
        video: '',
      },
      date: 'September 16, 2021',
      commentCount: 30,
      likeCount: 8750,
      avatar: 'AB',
      timeStamp: {},
      comments: [
        {
          commentId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
          userId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
          content: 'somewhat random text',
          timeStamp: {},
          parentPostId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
        },
      ],
      flagedBy: [
        'd290f1ee-6c54-4b01-90e6-d701748f0851',
      ],
      hiddenBy: [
        'd290f1ee-6c54-4b01-90e6-d701748f0851',
      ],
    },
    {
      postId: 'd290f1ee-6c54-4b01-90e6-d701748f0852',
      groupId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
      userId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
      content: {
        title: 'Look at the Galaxy!',
        text: 'beautiful night',
        image: '/galaxyPic.jpeg',
        audio: '',
        video: '',
      },
      timeStamp: {},
      date: 'September 17, 2021',
      commentCount: 328,
      likeCount: 18650,
      avatar: 'NK',
      comments: [
        {
          commentId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
          userId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
          content: 'somewhat random text',
          timeStamp: {},
          parentPostId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
        },
      ],
      flagedBy: [
        'd290f1ee-6c54-4b01-90e6-d701748f0851',
      ],
      hiddenBy: [
        'd290f1ee-6c54-4b01-90e6-d701748f0851',
      ],
    },
  ],
  groupName: 'Galaxy',
  userStatus: 'joined',
};

export const listOfAdmins = [
  {
    userId: '11',
    userName: 'Alex',
    profilePicture: '',
  },
  {
    userId: '12',
    userName: 'Tom',
    profilePicture: '',
  },
];

export const listOfGroupAll = [
  {
    groupId: '11',
    groupName: 'Galaxy',
    groupPic: '',
    member: '40',
    userStatus: 'joined',
  },
  {
    groupId: '2',
    groupName: 'SEPTA',
    groupPic: '',
    member: '40',
    userStatus: 'not joined',
  },
  {
    groupId: '3',
    groupName: 'Study Group',
    groupPic: '',
    member: '48920',
    userStatus: 'joined',
  },
  {
    groupId: '4',
    groupName: 'Involution',
    groupPic: '',
    member: '345781',
    userStatus: 'pending',
  },
  {
    groupId: '5',
    groupName: 'Ubuntu',
    groupPic: '',
    member: '908907',
    userStatus: 'pending',
  },
  {
    groupId: '6',
    groupName: 'Debian',
    groupPic: '',
    member: '462',
    userStatus: 'not joined',
  },
  {
    groupId: '7',
    groupName: 'CentOS',
    groupPic: '',
    member: '471',
    userStatus: 'joined',
  },
  {
    groupId: '8',
    groupName: 'Assembly',
    groupPic: '',
    member: '589',
    userStatus: 'not joined',
  },
  {
    groupId: '9',
    groupName: 'JIT',
    groupPic: '',
    member: '25',
    userStatus: 'not joined',
  },
  {
    groupId: '10',
    groupName: 'NLP',
    groupPic: '',
    member: '5',
    userStatus: 'not joined',
  },
];

export const listOfGroupPub = [
  {
    groupId: '11',
    groupName: 'Galaxy',
    groupPic: '',
    groupInfo: { createOn: '', member: '40' },
    userStatus: 'joined',
  },
  {
    groupId: '2',
    groupName: 'SEPTA',
    groupPic: '',
    groupInfo: { createOn: '', member: '3640' },
    userStatus: 'not join',
  },
  {
    groupId: '3',
    groupName: 'Study Group',
    groupPic: '',
    groupInfo: { createOn: '', member: '48920' },
    userStatus: 'joined',
  },
  {
    groupId: '4',
    groupName: 'Involution',
    groupPic: '',
    groupInfo: { createOn: '', member: '345781' },
    userStatus: 'pending',
  },
  {
    groupId: '5',
    groupName: 'Ubuntu',
    groupPic: '',
    groupInfo: { createOn: '', member: '908907' },
    userStatus: 'pending',
  },
  {
    groupId: '6',
    groupName: 'Debian',
    groupPic: '',
    groupInfo: { createOn: '', member: '462' },
    userStatus: 'notJoined',
  },
];
