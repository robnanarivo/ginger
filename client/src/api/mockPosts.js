const postData = {
  topics: ['Sports', 'Universe', 'Science', 'Astronomy'],
  groupInfo: { createOn: 'Feb 20 2008', member: '2.8k' },
  groupId: '11',
  posts: [
    {
      postId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
      groupId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
      creatorId: '61b26b150c01529bed74c6f0',
      creatorName: 'Wang',
      title: 'Stars are so beautiful',
      content: {
        text: 'star!',
        image: '/starPic.jpeg',
        audio: '',
        video: '',
      },
      profilePicture: '/starPic.jpeg',
      timestamp: 1639429592291,
      comments: [
        {
          commentId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
          userId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
          content: 'somewhat random text',
          timeStamp: 1639429592456,
          parentPostId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
        },
      ],
      flagedBy: [
        'd290f1ee-6c54-4b01-90e6-d701748f0851',
      ],
      hiddenBy: [
        'd290f1ee-6c54-4b01-90e6-d701748f0851',
      ],
    },
  ],
  groupName: 'Galaxy',
  userStatus: 'joined',
};

export default postData;
