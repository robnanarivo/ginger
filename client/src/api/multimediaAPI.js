import axios from 'axios';
import domain from '../constants/profile';

const upload = async (file) => {
  try {
    const formData = new FormData();
    formData.append('file', file);
    const res = await axios.post(`${domain}/api/multimedia`, formData, {
      headers: {
        'Content-Type': 'multipart/form-data',
      },
    });
    return res.data;
  } catch (err) {
    return err.response;
  }
};

export default upload;
