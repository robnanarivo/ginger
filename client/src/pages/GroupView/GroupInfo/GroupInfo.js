import React from 'react';
import Box from '@mui/material/Box';
import { useHistory, useParams } from 'react-router-dom';
import Typography from '@mui/material/Typography';
import Button from '@mui/material/Button';
import Divider from '@mui/material/Divider';
import TopicList from './TopicList';
import Stats from './Stats';
import { updateGroup } from '../../../api';

function displayJoinButton(updateUS, userStatus, groupId) {
  if (userStatus === 'not joined') {
    return (
      <Button
        variant="contained"
        onClick={() => {
          const res = updateGroup(groupId, 'apply', [sessionStorage.getItem('userId')], []);
          if (res) {
            updateUS('pending');
          }
        }}
        sx={{
          width: 120,
          height: 30,
          borderRadius: 8,
          fontWeight: 'bold',
        }}
      >
        JOIN
      </Button>
    );
  }
  if (userStatus === 'pending') {
    return (
      <Button
        variant="contained"
        onClick={() => {
          const res = updateGroup(groupId, 'leave', [sessionStorage.getItem('userId')], []);
          if (res) {
            updateUS('not joined');
          }
        }}
        sx={{
          width: 120,
          height: 30,
          borderRadius: 8,
          fontWeight: 'bold',
          color: '#AFAFAF',
          backgroundColor: '#E0E0E0',
          '&:hover': {
            cursor: 'pointer',
            backgroundColor: '#C0C0C0',
          },
        }}
      >
        PENDING
      </Button>
    );
  }
  return (<div />);
}

export default function GroupInfo(props) {
  const {
    topics,
    groupInfo,
    userStatus,
    updateUserStatus,
    inviteUserPop,
    isAdmin,
  } = props;
  const history = useHistory();
  const { groupId } = useParams();
  return (
    <Box
      sx={{
        bgcolor: '#FFFFFF',
        borderRadius: 2,
        width: '300px',
      }}
    >
      <Typography variant="h5" sx={{ marginLeft: '8%' }}> About </Typography>
      <Divider sx={{ backgroundColor: 'black' }} />
      <Box sx={{ marginLeft: '8%', marginTop: '4%' }}>
        <Typography variant="h6"> Topics </Typography>
        <TopicList topics={topics} sx={{ width: '80%' }} />
        <Typography variant="h6" sx={{ marginTop: '8%' }}> Statistics </Typography>
        <Stats groupInfo={groupInfo} />
      </Box>
      <Box
        sx={{
          display: 'flex',
          justifyContent: 'center',
          marginTop: '8%',
          height: 60,
        }}
      >
        { (userStatus !== 'joined')
          ? displayJoinButton(updateUserStatus, userStatus, groupId)
          : (
            <Button
              variant="contained"
              onClick={inviteUserPop}
              sx={{
                width: 120,
                height: 30,
                borderRadius: 8,
                fontWeight: 'bold',
              }}
            >
              INVITE
            </Button>
          )}
        <Box sx={{ marginLeft: 1, marginRight: 2 }} />
        {
          isAdmin && userStatus === 'joined'
            ? (
              <Button
                variant="contained"
                onClick={() => history.push(`${history.location.pathname}/adminPanel`)}
                sx={{
                  width: 120,
                  height: 30,
                  borderRadius: 8,
                  fontWeight: 'bold',
                }}
              >
                Manage
              </Button>
            )
            : (
              <div />
            )
        }
      </Box>
    </Box>
  );
}
