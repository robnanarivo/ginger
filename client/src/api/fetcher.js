// import { groupData, listOfGroupAll, listOfGroupPub } from './mockFetch';
import axios from 'axios';
import { listOfAdmins } from './mockFetch';

const domain = !process.env.NODE_ENV || process.env.NODE_ENV === 'development'
  ? 'http://localhost:5000'
  : '';

const getGroupData = async (groupId) => {
  const token = sessionStorage.getItem('userToken');
  try {
    const config = {
      authorization: `Bearer ${token}`,
    };
    const ret = await axios.get(`${domain}/api/group/${groupId}`, { headers: config });
    const { data } = ret;
    if (data.groupIcon !== '') {
      Object.assign(data, { groupIcon: `${domain}/api/multimedia/${data.groupIcon}` });
    } else {
      Object.assign(data, { groupIcon: '' });
    }
    // data.posts.forEach((post) => {
    //   const { content } = post;
    //   if (content.audio) {
    //     Object.assign(content, { audio: `${domain}/api/multimedia/${content.audio}` });
    //   }
    //   if (content.video) {
    //     Object.assign(content, { video: `${domain}/api/multimedia/${content.video}` });
    //   }
    //   if (content.image) {
    //     Object.assign(content, { image: `${domain}/api/multimedia/${content.image}` });
    //   }
    // });
    return data;
  } catch (err) {
    return err.response;
  }
};

const createGroup = async (groupInfo) => {
  const token = sessionStorage.getItem('userToken');
  try {
    const config = {
      authorization: `Bearer ${token}`,
    };
    const res = await axios.post(`${domain}/api/group/`, groupInfo, { headers: config });
    return res.data;
  } catch (err) {
    // console.log(err);
    return err.response.data;
  }
};

const updateGroup = async (groupId, action, userIds, adminIds) => {
  const token = sessionStorage.getItem('userToken');
  const updateInfo = { action, userIds, adminIds };
  try {
    const config = {
      authorization: `Bearer ${token}`,
    };
    const res = await axios.put(`${domain}/api/group/${groupId}`, updateInfo, { headers: config });
    // console.log(res);
    if (res.data.results === 'success') {
      return true;
    }
    return false;
  } catch (err) {
    // console.log(err);
    return false;
  }
};

const getListOfGroups = async (topic) => {
  const token = sessionStorage.getItem('userToken');
  try {
    const config = {
      authorization: `Bearer ${token}`,
    };
    const ret = await axios.get(`${domain}/api/groups/${topic}`, { headers: config });
    const { data } = ret;
    data.forEach((d) => {
      if (d.groupIcon !== '') {
        Object.assign(d, { groupIcon: `${domain}/api/multimedia/${d.groupIcon}` });
      } else {
        Object.assign(d, { groupIcon: '' });
      }
    });
    return ret.data;
  } catch (err) {
    // console.log(err.response);
    return err.response;
  }
};

// const getGroupAdmins = async (groupId) => {
//   const groupInfo = ['Alex', 'Tom'];
//   if (groupId) {
//     return listOfAdmins;
//   }
//   // const token = sessionStorage.getItem('userToken');
//   // try {
//   //   const config = {
//   //     authorization: `Bearer ${token}`,
//   //   };
//   //   const ret = await axios.get(`${domain}/api/group/${groupId}`, { headers: config });
//   //   const { data } = ret;
//   //   console.log(data);
//   // } catch (err) {
//   //   return err.response;
//   // }
//   return groupInfo;
// };

const getGroupAdmins = async (groupId) => {
  const groupInfo = ['Alex', 'Tom'];
  if (groupId) {
    return listOfAdmins;
  }
  // const token = sessionStorage.getItem('userToken');
  // try {
  //   const config = {
  //     authorization: `Bearer ${token}`,
  //   };
  //   const ret = await axios.get(`${domain}/api/group/${groupId}`, { headers: config });
  //   const { data } = ret;
  //   console.log(data);
  // } catch (err) {
  //   return err.response;
  // }
  return groupInfo;
};

const removeAdmin = async (adminId) => {
  if (!adminId) {
    return true;
  }
  return true;
};

export {
  getGroupData,
  getListOfGroups,
  createGroup,
  updateGroup,
  getGroupAdmins,
  removeAdmin,
};
