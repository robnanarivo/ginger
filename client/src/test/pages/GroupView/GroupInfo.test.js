import React from 'react';
import renderer from 'react-test-renderer';
import {
  screen,
  render,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import GroupInfo from '../../../pages/GroupView/GroupInfo';
import { updateGroup } from '../../../api';
import 'jest-styled-components';

jest.mock('../../../api');

describe('GroupInfo unit test', () => {
  beforeEach(() => {
    updateGroup.mockResolvedValue(true);
  });

  it('GroupInfo Snapshot test', async () => {
    const history = createMemoryHistory();
    let tree;
    await renderer.act(async () => {
      tree = renderer.create(
        <Router history={history}>
          <GroupInfo
            topics={['Sports', 'Universe', 'Science']}
            groupInfo={{ createOn: 'Feb 20 2008', member: '2.8k' }}
            updateUserStatus={() => {}}
            userStatus="joined"
            inviteUserPop={() => {}}
            isAdmin={false}
          />
        </Router>,
      );
    });
    expect(tree).toMatchSnapshot();
  });

  it('GroupInfo contains correct element joined', async () => {
    const history = createMemoryHistory();
    await act(async () => {
      render(
        <Router history={history}>
          <GroupInfo
            topics={['Sports', 'Universe', 'Science']}
            groupInfo={{ createOn: 'Feb 20 2008', member: '2.8k' }}
            updateUserStatus={() => {}}
            userStatus="joined"
            inviteUserPop={() => {}}
            isAdmin={false}
          />
        </Router>,
      );
    });
    // contains group info heading
    expect(screen.getByText('About')).toBeInTheDocument();
    // contains leave button
    expect(screen.getByText('INVITE')).toBeInTheDocument();
  });

  it('GroupInfo contains correct element not joined', async () => {
    const history = createMemoryHistory();
    await act(async () => {
      render(
        <Router history={history}>
          <GroupInfo
            topics={['Sports', 'Universe', 'Science']}
            groupInfo={{ createOn: 'Feb 20 2008', member: '2.8k' }}
            updateUserStatus={() => {}}
            userStatus="not joined"
            inviteUserPop={() => {}}
            isAdmin={false}
          />
        </Router>,
      );
    });
    // contains group info heading
    expect(screen.getByText('About')).toBeInTheDocument();
    // contains join button
    expect(screen.getByText('JOIN')).toBeInTheDocument();
    const button = screen.getByText('JOIN');
    // click button
    await act(async () => {
      userEvent.click(button);
    });
  });

  it('GroupInfo contains correct element pending', async () => {
    const history = createMemoryHistory();
    await act(async () => {
      render(
        <Router history={history}>
          <GroupInfo
            topics={['Sports', 'Universe', 'Science']}
            groupInfo={{ createOn: 'Feb 20 2008', member: '2.8k' }}
            updateUserStatus={() => {}}
            userStatus="pending"
            inviteUserPop={() => {}}
            isAdmin={false}
          />
        </Router>,
      );
    });
    // contains group info heading
    expect(screen.getByText('About')).toBeInTheDocument();
    // contains join button
    expect(screen.getByText('PENDING')).toBeInTheDocument();
    const button = screen.getByText('PENDING');
    // click button
    await act(async () => {
      userEvent.click(button);
    });
  });
});
