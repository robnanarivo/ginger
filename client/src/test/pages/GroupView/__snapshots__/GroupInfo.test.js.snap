// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupInfo unit test GroupInfo Snapshot test 1`] = `
<div
  className="MuiBox-root css-1c5he4i"
>
  <h5
    className="MuiTypography-root MuiTypography-h5 css-5th0l0-MuiTypography-root"
  >
     About 
  </h5>
  <hr
    className="MuiDivider-root MuiDivider-fullWidth css-fkcljs-MuiDivider-root"
  />
  <div
    className="MuiBox-root css-dto4a0"
  >
    <h6
      className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
       Topics 
    </h6>
    <div
      className="MuiBox-root css-1290pys"
    >
      <div
        className="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1dzpuen-MuiChip-root"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
      >
        <span
          className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
        >
          Sports
        </span>
      </div>
      <div
        className="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1dzpuen-MuiChip-root"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
      >
        <span
          className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
        >
          Universe
        </span>
      </div>
      <div
        className="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1dzpuen-MuiChip-root"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
      >
        <span
          className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
        >
          Science
        </span>
      </div>
    </div>
    <h6
      className="MuiTypography-root MuiTypography-h6 css-1bywiql-MuiTypography-root"
    >
       Statistics 
    </h6>
    <div
      className="css-12qb12d-MuiStack-root"
    >
      <div
        className="MuiBox-root css-0"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-ojz4j6-MuiTypography-root"
        >
           Created on
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Feb 20 2008
        </p>
      </div>
      <div
        className="MuiBox-root css-0"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-ojz4j6-MuiTypography-root"
        >
           Members
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          2.8k
        </p>
      </div>
    </div>
  </div>
  <div
    className="MuiBox-root css-1prb51h"
  >
    <button
      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-92u5jt-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      INVITE
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <div
      className="MuiBox-root css-bjhy15"
    />
    <div />
  </div>
</div>
`;
